### DISCUSSION TITLE
Sub upgrade for cameras doesn't work with a camera cycling system

### DISCUSSION THREAD

**Discussion by Petralicious1** [FakeFishGames/Barotrauma#13603] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

I have created a custom sub with two types of cameras, which is being used to play a campaign:

- One is a fixed surveillance camera outside the airlock, wired directly to a periscope; this one works as expected in every way, including the sub upgrade drastically increasing it's viewing range.

- Another is a cycling camera system which is wired as follows:
- - Periscope -> Wifi Component -> Camera
- - The wifi component receives a siglan in SET_CHANNEL from a simple counter that increments from 1 to 6 by one step every time the trigger of the periscope is pressed, and returns to 1 afterwards, cycling which wifi it's position signal is sent to, effectively swapping the cameras.

The latter's viewing range is not increased at all by sub upgrades for some reason.

### Reproduction steps

1. Build a sub with a camera switching system as described. (optional: add a normally, directly wired camera for comparison)
2. Load it in a campaign
3. Upgrade the cameras
4. Use the periscopes to see the difference between camera viewing ranges

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.2.8.0 (Winter Update hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
The upgrade increases the view ranges of turrets, not cameras.
